core/src/test/java/org/pentaho/di/connections/vfs/provider/BaseVFSConnectionProviderTest.java

```diff
@@ -0,0 +1,85 @@
+import org.pentaho.di.core.exception.KettleException;
+
+import static org.mockito.Mockito.when;
+
+public class BaseVFSConnectionProviderTest {
+  @Mock
+  BaseVFSConnectionDetails baseVFSConnectionDetails;
+
+  @Before
+  public void setup() {
+    baseVFSConnectionDetails = Mockito.spy( BaseVFSConnectionDetails.class );
+  }
+
+  @Test
+  public void test1() throws KettleException {
+
+    BaseVFSConnectionProvider<VFSConnectionDetails> baseVFSConnectionProvider = Mockito.mock( BaseVFSConnectionProvider.class, Answers.CALLS_REAL_METHODS );
+
+    when( baseVFSConnectionProvider.test( baseVFSConnectionDetails ) ).thenReturn( Boolean.TRUE );
+    when( baseVFSConnectionDetails.getRootPath() ).thenReturn( "xyz/" );
+    Assert.assertFalse( baseVFSConnectionProvider.test( baseVFSConnectionDetails, getVFSTestOptionWhenNotIgnoringRootPath() ) );
+  }
+
+  @Test
+  public void test2() throws KettleException {
+
+    BaseVFSConnectionProvider<VFSConnectionDetails> baseVFSConnectionProvider = Mockito.mock( BaseVFSConnectionProvider.class, Answers.CALLS_REAL_METHODS );
+    Assert.assertFalse( baseVFSConnectionProvider.test( baseVFSConnectionDetails, getVFSTestOptionWhenNotIgnoringRootPath() ) );
+  }
+
+
+  @Test
+  public void test3() throws KettleException {
+
+    BaseVFSConnectionProvider<VFSConnectionDetails> baseVFSConnectionProvider = Mockito.mock( BaseVFSConnectionProvider.class, Answers.CALLS_REAL_METHODS );
+
+    when( baseVFSConnectionProvider.test( baseVFSConnectionDetails ) ).thenReturn( Boolean.TRUE );
+    when( baseVFSConnectionDetails.isSupportsRootPath() ).thenReturn( Boolean.FALSE );
+    Assert.assertTrue( baseVFSConnectionProvider.test( baseVFSConnectionDetails, getVFSTestOptionWhenIgnoringRootPath() ) );
+  }
+
+  @Test
+  public void test4() throws KettleException {
+
+    BaseVFSConnectionProvider<VFSConnectionDetails> baseVFSConnectionProvider = Mockito.mock( BaseVFSConnectionProvider.class, Answers.CALLS_REAL_METHODS );
+
+    when( baseVFSConnectionProvider.test( baseVFSConnectionDetails ) ).thenReturn( Boolean.TRUE );
+    Assert.assertTrue( baseVFSConnectionProvider.test( baseVFSConnectionDetails, getVFSTestOptionWhenNotIgnoringRootPath() ) );
+  }
+
+
+  @Test
+  public void test5() throws KettleException {
+
+    BaseVFSConnectionProvider<VFSConnectionDetails> baseVFSConnectionProvider = Mockito.mock( BaseVFSConnectionProvider.class, Answers.CALLS_REAL_METHODS );
+
+    when( baseVFSConnectionProvider.test( baseVFSConnectionDetails ) ).thenReturn( Boolean.TRUE );
+    when( baseVFSConnectionDetails.getRootPath() ).thenReturn( "xyz/" );
+    when( baseVFSConnectionDetails.getDomain() ).thenReturn( "local" );
+    Assert.assertFalse( baseVFSConnectionProvider.test( baseVFSConnectionDetails, getVFSTestOptionWhenNotIgnoringRootPath() ) );
+  }
+
+  private VFSConnectionTestOptions getVFSTestOptionWhenNotIgnoringRootPath() {
+    return new VFSConnectionTestOptions( Boolean.FALSE );
```

Comment:
Why use these `Boolean.FALSE` and true constants, here and elsewhere in this file?